question22/main.js

```diff
@@ -0,0 +1,188 @@
+'use strict';
+const wrapper = document.getElementById('js-wrapper');
+
+const createElementWithClassName = (element, name) => {
+  const createdElement = document.createElement(element);
+  createdElement.classList.add(name);
+  return createdElement;
+};
+
+const showLoadImg = () => {
+  const loadImg = createElementWithClassName("img", "loading");
+  loadImg.src = "./img/loading-circle.gif";
+  loadImg.id = 'js-loading';
+  wrapper.appendChild(loadImg);
+} 
+
+const removeLoadImg = () => document.getElementById('js-loading').remove();
+
+const getData = new Promise((resolve) => {
+  showLoadImg();
+  setTimeout(() => {
+    resolve(fetch('https://mocki.io/v1/5381547b-04ff-4a89-b169-dd84425f5196'));
+  }, 3000);
+})
+
+//Get json data
+const callApi = async() => {
+  try {
+    const response = await getData;
+    if (!response.ok) {
+      console.error(`${response.status}:${response.statusText}`);
+    }
+    const json = await response.json();
+    return json.data;
+  }
+  catch(error) {
+    console.error(error);
+  }
+  finally {
+    removeLoadImg();
+  }
+}
+
+const init = async() => {
+  const userData = await callApi();
+  if (userData){
+    renderTableElement(userData);
+    } else {
+    wrapper.textContent = "Failed to display data.";
+    console.error('Failed to display data.'); 
+  }
+}
+init();
+
+const userDataColumn = {
+  userId: {
+    value : "ID",
+    sort : true
+  },
+  name: {
+    value : "名前",
+    sort : false
+  },
+  gender: {
+    value : "性別",
+    sort : false
+  },
+  age: {
+    value : "年齢",
+    sort : true
+  }
+}
```

Comment:
Specify here whether or not sorting is required.

Reply:
IMO
I received a review from Moriken-sensei here the other day.

I heard that by using `has**` you can explicitly indicate that a boolean value is included.
```suggestion
const userDataColumn = {
  userId: {
    value : "ID",
    hasSort : true
  },
  name: {
    value : "名前",
    hasSort : false
  },
  gender: {
    value : "性別",
    hasSort : false
  },
  age: {
    value : "年齢",
    hasSort : true
  }
}
```

Reply:
Thank you for sharing that information! I learned a lot. I will make use of it in the future!
fixed commit : https://github.com/csakyo/homework/pull/37/commits/3bd79a695dbd9436b126b2e91116215a04a1bfdd